### PR TITLE
add chat-engine and identity docs

### PR DESCRIPTION
**Pull request by BolaNasr**

### CODE REVIEW

File: docs/wiki/identity.md [hunk 1]
```diff
@@ -18,7 +18,10 @@ me.tid
 # or
 j.core.identity.me.tid
 ```
-
+To register your identity in TFGrid
+```python
+me.register()
```
Comment (ranatrk) [inline docs/wiki/identity.md:23]: isn't it `j.core.identity.me.register()`?

File: docs/wiki/chat_engine.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+- download_file
+- upload_file
+- qrcode_show
+- captcha_ask
+- md_show
+- md_show_confirm
+- loading_show
+- new_form
+- multi_values_ask
+- chatflow_step
```
Comment (waleedhammam) [inline docs/wiki/chat_engine.md:81]: not a type, it's a decorator for the chatflow slide step

File: docs/wiki/chat_engine.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
+# Chatflow Engine
+
+
+Chatflows are an interactive communication bot that can be used to interactively ask the user some questions then perform actions based on the user's choices. think of serving a customer, handling invoices, buying items or in our case provisioning workloads on the grid.
```
Comment (ranatrk) [inline docs/wiki/chat_engine.md:4]: Think instead of think?

File: docs/wiki/chat_engine.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+## Usage
+
+it's very easy to define a new bot, you just need to make sure it's added in a package and that package is installed. The chatflow should be under the `/chatflows` directory in the package created.
```
Comment (ranatrk) [inline docs/wiki/chat_engine.md:8]: It's ?

Comment (ranatrk) [reply docs/wiki/chat_engine.md:8]: and it is under `/chats` not `/chatflows`

File: docs/wiki/identity.md [hunk 5]
```diff
@@ -18,7 +18,10 @@ me.tid
...
+```
```
Comment (waleedhammam) [inline docs/wiki/identity.md:24]: add me.save() to save the instance in filesystem

File: docs/wiki/chat_engine.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            "sides": ["fries", "Onion rings"],
+        },
+        "3 Pizza": {
+            "main": ["Chicken Pizza", "Beef Pizza", "Cheese Pizza"],
+            "sides": ["fries", "Cheese"],
+        }
+    }
+
+    # Ask the user about his name
+    name = self.string_ask("Hello, What's your name?")
```
Comment (ranatrk) [inline docs/wiki/chat_engine.md:31]: Can you add default as an example?

File: docs/wiki/chat_engine.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    # display a dropdown containing your favourite Restaurants
+    restaurant_name = self.drop_down_choice("Please select a Resturant", menus.keys())
+
+    # display the main dishes of the selected restaurant so the user can choose only one dish
+    main_dish = self.single_choice("Please Select your main dish", menus[restaurant_name]["main"])
+
+    # ask about the mount (this accepts any integer)
+    amount = self.int_ask("How many {} do you want".format(main_dish))
+
+    # ask about the side dishes (the user can choose multible side dishes)
```
Comment (ranatrk) [inline docs/wiki/chat_engine.md:42]: multiple instead of multible

File: docs/wiki/identity.md (hunk 1 above)
Comment (BolaNasr) [reply docs/wiki/identity.md:23]: ```me = j.core.identity.new("name", "threebot_name", "threebot_mail", "words") ```

**Pull request merged**